pkg/broker/fanout/stats_reporter.go

```diff
@@ -0,0 +1,161 @@
+package fanout
```

Comment:
I wonder if it makes sense to have all stats_reporters in sub-packages of metrics. This looks more like a "delivery stats reporter". We can put this under `metrics/delivery/stats_reporter`. Similarly for ingress. Thoughts?

Reply:
How about pkg/observability/delivery_stats.go or pkg/metrics/delivery_stats.go? I think a package per reporter might be too much.

Reply:
I don't have strong preference. I usually do different package names with the same struct name (like I did for the sync pool), but I'm fine with this approach too.

Reply:
I'll do pkg/metrics/delivery_stats.go for now, mainly because we can create subdirectories later if there ends up being a lot of files in pkg/metrics in the future

Reply:
I ended up doing @yolocs suggestion since the both the ingress's and delivery's StatsReporter had the same function and metric names which caused a conflict being in the same package. Ending up with:

`pkg/metrics/delivery/stats_reporter.go`
     And going to create a follow up PR to move the ingress metrics to
`pkg/metrics/ingress/stats_reporter.go`

Reply:
You can rename the variables or make them a field in the struct.  In fact, @ian-mi just did this for ingress recently, if you sync with master. What about `metrics/broker_stats_reporter` and `metrics/trigger_stats_reporter`?

Reply:
Sync'd with @liu-cong offline and ended up renaming and moving some fields into the struct. Now everything's in `pkg/metrics` like @ian-mi suggested